src/components/NewAppModal.js

```diff
@@ -0,0 +1,161 @@
+import { div, h, li, span, ul } from 'react-hyperscript-helpers'
+import { ButtonPrimary, Link } from 'src/components/common'
+import { icon } from 'src/components/icons'
+import { withModalDrawer } from 'src/components/ModalDrawer'
+import TitleBar from 'src/components/TitleBar'
+import { machineTypes } from 'src/data/machines'
+import { Ajax } from 'src/libs/ajax'
+import { currentApp } from 'src/libs/cluster-utils'
+import colors from 'src/libs/colors'
+import { withErrorReporting } from 'src/libs/error'
+import * as Style from 'src/libs/style'
+import * as Utils from 'src/libs/utils'
+
+
+const styles = {
+  whiteBoxContainer: { padding: '1rem', borderRadius: 3, backgroundColor: 'white' },
+  drawerContent: { display: 'flex', flexDirection: 'column', flex: '0 1 0%', padding: '0.5rem 1.5rem' },
+  headerText: { fontSize: 16, fontWeight: 600 }
+}
+
+// TODO: Once we have more appTypes, abstract text/component to support other apps
+export const NewAppModal = _.flow(
+  Utils.withDisplayName('NewAppModal'),
+  withModalDrawer({ width: 675 })
+)(({ onDismiss, onSuccess, namespace, apps, bucketName, workspaceName }) => {
+  const [viewMode, setViewMode] = useState(undefined)
+
+  const app = currentApp(apps)
+
+  const createApp = withErrorReporting('Error creating app', async () => {
+    await Ajax().Apps.app(namespace, Utils.generateKubernetesClusterName()).create(Utils.generatePersistentDiskName(), 'GALAXY', namespace, bucketName, workspaceName)
+    return onSuccess()
+  })
+
+  const deleteApp = withErrorReporting('Error deleting app', async () => {
+    await Ajax().Apps.app(app.googleProject, app.appName).delete()
+    return onSuccess()
+  })
+
+  const renderActionButton = () => {
+    return Utils.switchCase(viewMode,
+      ['deleteWarn', () => {
+        return h(ButtonPrimary, { onClick: () => deleteApp() }, ['Delete'])
+      }],
+      ['createWarn', () => {
+        return h(ButtonPrimary, { onClick: () => createApp() }, ['Create'])
+      }],
+      [Utils.DEFAULT, () => {
+        return !!app ?
+          h(ButtonPrimary, { onClick: () => setViewMode('deleteWarn') }, ['Delete']) :
+          h(ButtonPrimary, { onClick: () => setViewMode('createWarn') }, ['Next'])
+      }]
+    )
+  }
+
+  const renderBottomButtons = () => {
+    return h(Fragment, [
+      div({ style: { display: 'flex', margin: '1rem 0 1rem' } }, [
+        div({ style: { flex: 1 } }),
+        renderActionButton()
+      ])
+    ])
+  }
+
+  const getMachineDetails = () => {
+    return _.filter(({ name }) => name === 'n1-standard-8', machineTypes)[0]
+  }
+
+  const renderCreateWarning = () => {
+    return h(Fragment, [
+      div({ style: { marginBottom: '1rem' } }, ['Environment will consist of an application and cloud compute.']),
+      div({ style: { ...styles.whiteBoxContainer, backgroundColor: colors.accent(.1), boxShadow: Style.standardShadow } }, [
+        div({ style: { display: 'flex', padding: `0.5rem 0 0`, minHeight: 120 } }, [
```

Comment:
nit: This wrapper could be removed, the `min-height` isn't having any effect and the padding looks extraneous to me.